vcr/issuer/keyresolver.go

```diff
@@ -0,0 +1,31 @@
+package issuer
+
+import (
+	"fmt"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/vdr/doc"
+	vdr "github.com/nuts-foundation/nuts-node/vdr/types"
+)
+
+// vdrKeyResolver resolves private keys based upon the VDR document resolver
+type vdrKeyResolver struct {
+	docResolver vdr.DocResolver
+	keyResolver crypto.KeyResolver
+}
+
+func (r vdrKeyResolver) ResolveAssertionKey(issuerDID did.DID) (crypto.Key, error) {
```

Comment:
No possibility to reuse the resolver in vdr/doc?

Reply:
The vdr does not return the private key. This is a combination of both the vdr and the private key store. It's goal is to encapsulate the key resolving proces for easier setup and testing.

Reply:
Feels a bit like one encapsulation too many. It's only used once, in `networkPublisher`. So you could just make it a member function of that type, and use VDR's `KeyResolver.ResolveAssertionKeyID` which does 80% of this function (so it's sort of duplicate code). I'm not convinced eliminating `KeyResolver.Resolve()` in the unit tests justifies this type.

Reply:
It's used in both the NetworkPublisher and in the BuildVC method. Both are different parts in the process. Later it will be used in the buildRevocation and PublishRevocation. I think it saves some duplicate code and surely saves some testing code, since we only need 1 test for a failed assertionKey lookup instead of a failed doc lookup and a failed keyID lookup and a failed keyLookup. Now the errors are equal over all these usages.